firewood/src/v2/api.rs

```diff
@@ -170,6 +171,40 @@ pub trait DbView {
         last_key: Option<K>,
         limit: Option<usize>,
     ) -> Result<Option<RangeProof<Vec<u8>, Vec<u8>>>, Error>;
+
+    /// Obtain a stream over the keys/values of this view, using an optional starting point
+    ///
+    /// # Arguments
+    ///
+    /// * `first_key` - If None, start at the lowest key
+    ///
+    /// # Note
+    ///
+    /// If you always want to start at the beginning, [DbView::iter] is easier to use
+    /// If you always provide a key, [DbView::iter_from] is easier to use
+    ///
+    async fn iter_option<K: KeyType>(
```

Comment:
Why is this async?

Reply:
Good question, it doesn't need to be! Will address in a followup.